src/Tools/dotnet-monitor/CollectionRules/CollectionRuleContainer.cs

```diff
@@ -54,11 +54,64 @@ public async ValueTask DisposeAsync()
         {
             if (DisposableHelper.CanDispose(ref _disposalState))
             {
-                _disposalTokenSource.SafeCancel();
+                await StopRulesCore(CancellationToken.None);
+            }
+        }
+
+        public async Task StartRulesAsync(
```

Comment:
I have no issues with this approach, but would it be easier to just destroy and recreate the entire rule container?

Reply:
To do that, the service would need to keep a list of IEndpointInfo/IProcessInfo objects in order to recreate the containers. 

Arguably, this might be necessary anyway since the service/containers/pipelines have no way of detecting when the target process has exited; either the container or the service needs to poll the process periodically and then shutdown the rules when the process has exited. I was thinking that maybe the container could do this polling and then notify the service when the process is gone; this way, all of the process specific logic is contained in the container.

I was hoping to tackle that in the next change.